oasis/models/rules.py

```
@@ -2,15 +2,167 @@
+            if key in Metrics:
+                val = Metrics[key]
+                if key not in self.cfg.metrics:
+                    continue
+
+                t = Thread(target=self.run_action,
+                           args=(key, val, self.cfg.metrics[key],
+                                 self.cfg.rules[key], ))
+                t.start()
+                self.threads[key] = t
+
+    def run_action(self, metric, val, config, rules):
+        logger.info("[job-id:{id}][metric:{metric}] start ot run"
+                    .format(id=sub_id(self.job.id), metric=metric))
+        self.compute(metric, val, config, rules)
+
+    def compute(self, metric, query_expr, config, rules):
+        while not self._exit:
+            now = datetime.datetime.now()
+            query = PrometheusQuery(query_expr,
+                                    time.mktime((now - datetime.timedelta(
+                                        seconds=timeparse(self.cfg.model["data_range"])))
+                                                .timetuple()),
+                                    time.mktime(now.timetuple()), "15s")
+            data_set = self.api.query(query)
+            if len(data_set) > 0:
+                values = []
+                for data in data_set.values():
+                    values.append(float(data))
+
+                features_value = {}
+                for key in config["features"]:
+                    if key in Features:
+                        features_value[key] = Features[key](values)
+
+                logger.info("[job-id:{id}][metric:{metric}] extraction features {value}, "
+                            "start to compare with rule"
+                            .format(id=sub_id(self.job.id),
+                                    metric=metric, value=features_value))
+
+                not_match_count = 0
+                for rule in rules:
+                    if rule["feature"] not in features_value:
+                        continue
```

I think we should report an error eagerly about this during initialization.